### PR TITLE
Minebots now check for pressure

### LINKED ISSUE #24136: Mine bots deal full (40) damage in pressurized environments
I haven't actually tested this on one of /tg/'s servers, but we ran into this issue after porting https://github.com/tgstation/tgstation/pull/19870 to Paradise, and as far as I can tell from the code this issue is also present here.

Because https://github.com/tgstation/tgstation/blob/master/code/modules/projectiles/guns/energy/kinetic_accelerator.dm#L155 is indented one step too far (so it's part of the kinetic_accelerator gun location check) the kinetic projectile deals its full forty damage when it's not fired from a kinetic accelerator (as is the case with mine bots) because the pressure check is never applied. 

If this is actually intended or not the case on the server do close this though - my bad in that case!

### PR DESCRIPTION
**Pull request by Robustin**
Fixes #24136

So apparently minebots do a flat 40 damage indoors, which meant an upgraded sentient minebot could crit people in under 3 seconds. The KA pressure check was in the ammo casing and simple mob code just poops out projectiles so... no pressure check. 

Thanks to Coiax for showing me prehit and letting me escape the nightmare of gun code.

### CODE REVIEW

**Comment by KingPhilipIII** [comment]
That explains how I saw a traitor hijack the shuttle with minebots...

File: code/modules/projectiles/guns/energy/kinetic_accelerator.dm [hunk 1]
```diff
@@ -138,30 +138,13 @@
...
 	projectile_type = /obj/item/projectile/kinetic
 	select_name = "kinetic"
 	e_cost = 500
 	fire_sound = 'sound/weapons/Kenetic_accel.ogg' // fine spelling there chap
 
-/obj/item/ammo_casing/energy/kinetic/ready_proj(atom/target, mob/living/user, quiet, zone_override = "")
-	..()
-	if(loc && istype(loc, /obj/item/weapon/gun/energy/kinetic_accelerator))
-		var/obj/item/weapon/gun/energy/kinetic_accelerator/KA = loc
-		KA.modify_projectile(BB)
```
Comment (ChangelingRain) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm]: important line!!!

Comment (Robustin) [inline code/modules/projectiles/guns/energy/kinetic_accelerator.dm]: ah, youre right

**Comment by Kromgar** [comment]
Still doesn't stop me from creating an army of evil minebots equipped with pressure mods MWAHAHAHAHAHAHA

**Comment by Robustin** [comment]
Don't see why it wouldnt, same code different spot. Wait can you load KA mods into the bots?

**Comment by Repukan** [comment]
You can't

**Comment by Kromgar** [comment]
Somrone said minebots could... maybe i was mistaken then

**Comment by AnonymousNow** [comment]
Oh hey, I was one of the minebots with that traitor! We really brutalised the crew with our guns, but hey - them's the boss' orders. They felt particularly powerful with our upgraded firing speed.

**Pull request merged**